data-layers/data-layer-api-caller/src/main/java/com/flipkart/aesop/apicallerdatalayer/upsert/ApiCallerUpsertDataLayer.java

```diff
@@ -0,0 +1,70 @@
+package com.flipkart.aesop.apicallerdatalayer.upsert;
+
+import com.flipkart.aesop.destinationoperation.UpsertDestinationStoreProcessor;
+import com.flipkart.aesop.event.AbstractEvent;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.linkedin.databus.client.pub.ConsumerCallbackResult;
+import org.json.JSONObject;
+import org.trpr.platform.core.impl.logging.LogFactory;
+import org.trpr.platform.core.spi.logging.Logger;
+
+import java.io.DataOutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Created by aman.gupta on 12/08/15.
+ */
+public class ApiCallerUpsertDataLayer extends UpsertDestinationStoreProcessor
+{
+    private static final Logger LOGGER = LogFactory.getLogger(ApiCallerUpsertDataLayer.class);
+    private URL url;
+    private JSONObject headers;
+    public ApiCallerUpsertDataLayer(URL url, JSONObject headers) {
+        this.url = url;
+        this.headers = headers;
+    }
+
+    @Override
+    protected ConsumerCallbackResult upsert(AbstractEvent event) {
+        try {
```

Comment:
Copy paste is very bad.
Abstract this out as a method in call. Instantiate the class every time. This will make it thread safe and there is no overhead of creating it over and over again. And then invoke the method.